sdk/src/main/java/com/mapbox/maps/plugin/MapPluginRegistry.kt

```diff
@@ -49,9 +49,11 @@ internal class MapPluginRegistry(
     mapInitOptions: MapInitOptions,
     plugin: Plugin
   ) {
-    plugin.instance?.let { mapPlugin ->
+    val mapPlugin = plugin.instance
```

Comment:
Previously `}?.let {...}` returned Unit, thus this [exception](https://github.com/mapbox/mapbox-maps-android/pull/1558/files#diff-770fab17e7054ea026d0d5ceda6083b2ef8895dcb103aadae1cd491afdea3fceL111) was never being thrown. Rewritten logic to be more readable and correct